cartographer_grpc/framework/rpc.cc

```diff
@@ -53,23 +74,53 @@ ::grpc::internal::ServerAsyncStreamingInterface* Rpc::streaming_interface() {
   LOG(FATAL) << "Never reached.";
 }
 
-Rpc::RpcState* Rpc::GetRpcState(State state) {
+::grpc::internal::AsyncReaderInterface<::google::protobuf::Message>*
+Rpc::async_reader_interface() {
+  switch (rpc_handler_info_.rpc_type) {
+    case ::grpc::internal::RpcMethod::CLIENT_STREAMING:
+      return server_async_reader_.get();
+    default:
+      LOG(FATAL) << "RPC type not implemented.";
+  }
+  LOG(FATAL) << "Never reached.";
+}
+
+Rpc::RpcState* Rpc::SetRpcStatePending(State state, bool pending) {
```

Comment:
Unusual to have a Set* method return a state. Perhaps there should be a separate get method.